contracts/eigenlayer/test/MirrorServiceHandler.t.sol

```diff
@@ -0,0 +1,413 @@
+        IWavsServiceHandler.SignatureData memory signatureData = createSignatureData(envelope, 4, 5);
+        
+        // Call handleSignedEnvelope should fail with invalid payload
+        vm.expectRevert();
+        serviceHandler.handleSignedEnvelope(envelope, signatureData);
+    }
+
+    function test_invalid_trigger_id() public {
+        // Create a valid UpdateWithId payload with triggerId = 2 (but we expect 1)
+        address[] memory newOperators = new address[](1);
+        address[] memory newSigningKeys = new address[](1);
+        uint256[] memory newWeights = new uint256[](1);
+        
+        newOperators[0] = address(0x123);
+        newSigningKeys[0] = address(0x456);
+        newWeights[0] = 10000;
+        
+        // Create the UpdateWithId struct with triggerId = 2
+        IMirrorUpdateTypes.UpdateWithId memory updateData = IMirrorUpdateTypes.UpdateWithId({
+            triggerId: 2,  // Invalid, should be 1
+            operators: newOperators,
+            signingKeys: newSigningKeys,
+            weights: newWeights
+        });
+        
+        // Create envelope with the encoded payload
+        IWavsServiceHandler.Envelope memory envelope = IWavsServiceHandler.Envelope({
+            eventId: bytes20(uint160(1)),
+            ordering: bytes12(0),
+            payload: abi.encode(updateData)
+        });
+        
+        // Create signature data with 4 operators (more than enough to pass quorum)
+        IWavsServiceHandler.SignatureData memory signatureData = createSignatureData(envelope, 4, 5);
+        
+        // Call handleSignedEnvelope should fail with InvalidTriggerId
+        vm.expectRevert(abi.encodeWithSelector(IMirrorUpdateTypes.InvalidTriggerId.selector, 1));
+        serviceHandler.handleSignedEnvelope(envelope, signatureData);
+    }
+
+    function test_insufficient_quorum() public {
+        // Create a valid UpdateWithId payload with triggerId = 1
+        address[] memory newOperators = new address[](1);
+        address[] memory newSigningKeys = new address[](1);
+        uint256[] memory newWeights = new uint256[](1);
+        
+        newOperators[0] = address(0x123);
+        newSigningKeys[0] = address(0x456);
+        newWeights[0] = 10000;
+        
+        // Create the UpdateWithId struct with triggerId = 1
+        IMirrorUpdateTypes.UpdateWithId memory updateData = IMirrorUpdateTypes.UpdateWithId({
+            triggerId: 1,
+            operators: newOperators,
+            signingKeys: newSigningKeys,
+            weights: newWeights
+        });
+        
+        // Create envelope with the encoded payload
+        IWavsServiceHandler.Envelope memory envelope = IWavsServiceHandler.Envelope({
+            eventId: bytes20(uint160(1)),
+            ordering: bytes12(0),
+            payload: abi.encode(updateData)
+        });
+        
+        // Create signature data with only 3 operators (not enough for quorum)
+        // The quorum is 4/5 (80%) in the default setup
```

Comment:
Huh? ServiceManager quorum is 2/3 right? How do we update this?

Actually, there is no method to change that on the mirror, right? Please add a new ServiceHandler that can be used to sync the changes to quorumNumerator and quorumDenominator in the service manager. You should probably be able to just assign the handler as owner of the service manager to update the fraction needed.